src/CubedSpheres/conformal_cubed_sphere_grid.jl

```diff
@@ -186,42 +186,45 @@ end
 ##### Nodes for OrthogonalSphericalShellGrid
 #####
 
-@inline λnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Face,   LY::Face,   LZ) = @inbounds grid.λᶠᶠᵃ[i, j]
-@inline λnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Face,   LY::Center, LZ) = @inbounds grid.λᶠᶜᵃ[i, j]
-@inline λnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Center, LY::Face,   LZ) = @inbounds grid.λᶜᶠᵃ[i, j]
-@inline λnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Center, LY::Center, LZ) = @inbounds grid.λᶜᶜᵃ[i, j]
+const OSSG = OrthogonalSphericalShellGrid
 
-@inline φnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Face,   LY::Face,   LZ) = @inbounds grid.φᶠᶠᵃ[i, j]
-@inline φnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Face,   LY::Center, LZ) = @inbounds grid.φᶠᶜᵃ[i, j]
-@inline φnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Center, LY::Face,   LZ) = @inbounds grid.φᶜᶠᵃ[i, j]
-@inline φnode(i, j, k, grid::OrthogonalSphericalShellGrid, LX::Center, LY::Center, LZ) = @inbounds grid.φᶜᶜᵃ[i, j]
+@inline λnode(i, j, k, grid::OSSG, LX::Face,   LY::Face,   LZ; kwargs...) = @inbounds grid.λᶠᶠᵃ[i, j]
+@inline λnode(i, j, k, grid::OSSG, LX::Face,   LY::Center, LZ; kwargs...) = @inbounds grid.λᶠᶜᵃ[i, j]
+@inline λnode(i, j, k, grid::OSSG, LX::Center, LY::Face,   LZ; kwargs...) = @inbounds grid.λᶜᶠᵃ[i, j]
+@inline λnode(i, j, k, grid::OSSG, LX::Center, LY::Center, LZ; kwargs...) = @inbounds grid.λᶜᶜᵃ[i, j]
```

Comment:
We don't want `kwargs` on low-level functions meant to be in-lined on the GPU (are kwargs supported in GPU kernels?)

Reply:
> We don't want `kwargs` on low-level functions meant to be in-lined on the GPU (are kwargs supported in GPU kernels?)

The GPU tests are passing (with the exception of two for now) so I think they're supported.

Can you explain why `kwargs` should be avoided? I haven't noticed any slowdowns because of that on my tests so I don't know what the downside is (although admittedly I haven't done a rigorous benchmarking routine yet).

Reply:
It may not matter here, but the compiler cannot specialize on the types of keyword arguments. The "Performance Tips" section of the julia docs used to recommend avoiding them, but I don't see a reference to that now, so the situation may be different...

Still it does not work with our function composition scheme. For example, we cannot take the difference of a node if kwargs are required.

I also don't understand the logic of the API --- why would users have to specify `with_halo=true` or `false` when asking for the node at index `i, j, k`? Does it just control whether a bounds error is thrown or not for the particular index?

Finally, I'm worried that if we do eventually generalize the code so that `xnodes` returns a field, this pattern will have to be refactored, again.

Reply:
> It may not matter here, but the compiler cannot specialize on the types of keyword arguments. 

I took that into consideration and, as it is now, things work as intended and specialize appropriately.

 
> Still it does not work with our function composition scheme. For example, we cannot take the difference of a node if kwargs are required. Composition of operators is the purpose of the `i, j, k, grid, args...` syntax pattern.
> 
> I also don't understand the logic of the API --- why would users have to specify `with_halo=true` or `false` when asking for the node at index `i, j, k`? Does it just control whether a bounds error is thrown or not for the particular index?

Yeah that was my intention. Although I see your point that this might not make much sense.

I'm focusing on get all tests passing for now and then I'll change the behavior. For now I'll change `xnode()` etc to not have any kwargs and alwats return the full array. Although maybe we can keep kwargs for `xnodes` for now and see how it looks before merging? Since this is more of a user-facing function I can see that being useful (e.g. per [the example that motivated this](https://github.com/CliMA/Oceananigans.jl/pull/2842/#discussion_r1065130759)).